src/sidebar/components/tag-editor.js

```diff
@@ -1,35 +1,202 @@
 'use strict';
 
-// @ngInject
-function TagEditorController(tags) {
-  this.onTagsChanged = function() {
-    tags.store(this.tagList);
-
-    const newTags = this.tagList.map(function(item) {
-      return item.text;
-    });
-    this.onEditTags({ tags: newTags });
+const { createElement } = require('preact');
+const propTypes = require('prop-types');
+const { useRef, useState } = require('preact/hooks');
+
+const { withServices } = require('../util/service-context');
+const SvgIcon = require('./svg-icon');
+
+/**
+ * Component to edit annotation tags. Tags may be added or removed from any
+ * given annotation. This component also renders suggested tags which use
+ * the `tags` service to set or get previously used tags to
+ * local storage.
+ */
+function TagEditor({ annotation, onEditTags, tags: tagsService, tagList }) {
+  const inputEl = useRef(null);
+  const [suggestions, setSuggestions] = useState([]);
+
+  /**
+   * Passes the tags list to the provided callback `onEditTags` so it may
+   * be saved to the service. Also passes the list to `tagsService.store`
+   * so it may be presented as a future suggestion if its not already
+   * one.
+   *
+   * @param {Array<string>} tagList
+   */
+  const saveTags = tagList => {
+    // save suggested tags
+    tagsService.store(tagList.map(tag => ({ text: tag })));
+    // set the new tag to the annotation
+    onEditTags({ tags: tagList });
   };
 
-  this.autocomplete = function(query) {
-    return Promise.resolve(tags.filter(query));
+  /**
+   * Remove a tag.
+   *
+   * @param {string} tag
+   */
+  const removeTag = tag => {
+    const newTagList = [...tagList]; // make a copy
+    const index = newTagList.indexOf(tag);
+    newTagList.splice(index, 1);
+    saveTags(newTagList);
+  };
+
+  /**
+   * Adds a new local tag equal to the value of the input field only
```

Comment:
Can you make this comment more clear? Also I think you can omit the bit about duplicate tags because you have a comment for it within the function.